pkg/controller/endpointslice/metrics/cache.go

```diff
@@ -128,34 +146,21 @@ type metricsUpdate struct {
 	desired, actual int
 }
 
-// desiredAndActualSlices returns a metricsUpdate with the desired and actual
-// number of EndpointSlices given the current values in the cache.
-// Must be called holding lock.
-func (c *Cache) desiredAndActualSlices() metricsUpdate {
-	mUpdate := metricsUpdate{}
-	for _, spCache := range c.cache {
```

Comment:
this was the problem right? we iterated ALL the cache, and now we store the values and update them accordingly.

Reply:
Yep, that's exactly what the problem was. Ended up not being very noticeable until you have a lot of Services, and most of the EndpointSlice scale tests focused on relatively small numbers of very large Services.